### PR TITLE
Fix conditional mark query Github failed issue

### PR DESCRIPTION
**Pull request by xwjiang-ms**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [x] 202012
- [x] 202205

#### What is the motivation for this PR?
Conditional mark failed to query github issue state to decide skip/xfail behaviors due to proxy issue
#### How did you do it?
Get proxies through inventory env.yml, and add proxies to requests

### CODE REVIEW

File: tests/common/plugins/conditional_mark/issue.py [hunk 1]
```diff
@@ -69,8 +71,29 @@ def is_active(self):
         Returns:
             bool: False if the issue is closed else True.
         """
+        INV_ENV_FILE = '../../../../ansible/group_vars/{}/env.yml'.format(self.inv_name)
+        PUBLIC_ENV_FILE = '../../../../ansible/group_vars/all/env.yml'
+        base_path = os.path.dirname(__file__)
+        if os.path.exists(INV_ENV_FILE):
+            env_file_path = os.path.join(base_path, INV_ENV_FILE)
+        else:
+            env_file_path = os.path.join(base_path, PUBLIC_ENV_FILE)
```
Comment (wangxin) [inline tests/common/plugins/conditional_mark/issue.py:80]: The env file could exists but have empty content (probably comment only). This logic here is not robust enough.

**Review by wangxin — CHANGES_REQUESTED** [review]

**Comment by wangxin** [comment]
After a second though, probably a more elegant way is to get http proxy in earlier stage and cache it. Then we do not need to pass down the inv_name argument a long way.

File: tests/common/plugins/conditional_mark/issue.py (hunk 1 above)
Comment (xwjiang-ms) [reply tests/common/plugins/conditional_mark/issue.py:79]: fixed

File: tests/common/plugins/conditional_mark/issue.py [hunk 2]
```diff
@@ -87,6 +109,27 @@ def is_active(self):
         return True
 
 
+def is_file_empty_or_comments(file_path):
```
Comment (wangxin) [inline tests/common/plugins/conditional_mark/issue.py:112]: If use `yaml.safe_load` to load a yaml file with only comments, will it raise exception?

File: tests/common/plugins/conditional_mark/issue.py [hunk 3]
```diff
@@ -69,8 +70,29 @@ def is_active(self):
         Returns:
             bool: False if the issue is closed else True.
         """
+        global g_inv_name
```
Comment (wangxin) [inline tests/common/plugins/conditional_mark/issue.py:73]: Is there a better way to parse proxy configuration and pass it down to the issue checker?

File: tests/common/plugins/conditional_mark/issue.py [hunk 4]
```diff
@@ -69,8 +70,29 @@ def is_active(self):
...
+        try:
+            with open(env_file_path) as env_file:
+                proxy_env = yaml.safe_load(env_file)
+                if proxy_env is not None:
+                    proxy = proxy_env.get("proxy_env", {})
+                    http_proxy = proxy.get('http_proxy', '')
+                    https_proxy = proxy.get('https_proxy', '')
+                    proxies = {'http_proxy': http_proxy, 'https_proxy': https_proxy}
+                else:
+                    proxies = {'http_proxy': '', 'https_proxy': ''}
```
Comment (wangxin) [inline tests/common/plugins/conditional_mark/issue.py:90]: The key in `proxies` should be `http` and `https` according to the python requests documentation.

Comment (xwjiang-ms) [reply tests/common/plugins/conditional_mark/issue.py]: modified

File: tests/common/plugins/conditional_mark/issue.py (hunk 2 above)
Comment (xwjiang-ms) [reply tests/common/plugins/conditional_mark/issue.py]: it won't, it will return None

File: tests/common/plugins/conditional_mark/issue.py (hunk 3 above)
Comment (xwjiang-ms) [reply tests/common/plugins/conditional_mark/issue.py]: fixed

File: tests/common/plugins/conditional_mark/__init__.py [hunk 5]
```diff
@@ -180,6 +180,49 @@ def get_basic_facts(session):
             session.config.cache.set('BASIC_FACTS', basic_facts)
 
 
+def is_file_empty_or_comments(file_path):
```
Comment (wangxin) [inline tests/common/plugins/conditional_mark/__init__.py:183]: Why not just take advantange of the `yaml.safe_load()` function?

Comment (xwjiang-ms) [reply tests/common/plugins/conditional_mark/__init__.py]: fixed

**Pull request closed without merging**